bokeh/models/sources.py

```diff
@@ -141,7 +142,7 @@ def from_df(cls, data):
             data (DataFrame) : data to convert
 
         Returns:
-            dict[str, list]
+            dict[str, np.array]
 
         '''
         return cls._data_from_df(data)
```

Comment:
I'd suggest a couple of tests, at a minimum we should maintain that passing a DataFrame results in a `.data` with `pd.Series` as column values